sdks/python/apache_beam/io/fileio.py

```diff
@@ -595,14 +602,21 @@ def __init__(self,
     self._compression_type = compression_type
 
     if self._readable():
-      self._decompressor = zlib.decompressobj(self._type_mask[compression_type])
+      if self._compression_type == CompressionTypes.BZIP2:
+        self._decompressor = bz2.BZ2Decompressor()
+      else:
+        self._decompressor = zlib.decompressobj(\
+                            self._type_mask[compression_type])
     else:
       self._decompressor = None
 
     if self._writeable():
-      self._compressor = zlib.compressobj(zlib.Z_DEFAULT_COMPRESSION,
-                                          zlib.DEFLATED,
-                                          self._type_mask[compression_type])
+      if self._compression_type == CompressionTypes.BZIP2:
+        self._compressor = bz2.BZ2Compressor(9)
```

Comment:
I recommend going with default compression level (ie don't specify one in the constructor).
Just bz2.BZ2Compressor()